pkg/crc/cluster/kubeadmin_password.go

```diff
@@ -57,12 +56,12 @@ func UpdateKubeAdminUserPassword(ocConfig oc.Config) error {
 	return ioutil.WriteFile(kubeAdminPasswordFile, []byte(kubeAdminPassword), 0600)
 }
 
-func GetKubeadminPassword(bundle *bundle.CrcBundleInfo) (string, error) {
+func GetKubeadminPassword() (string, error) {
 	kubeAdminPasswordFile := constants.GetKubeAdminPasswordPath()
 	rawData, err := ioutil.ReadFile(kubeAdminPasswordFile)
 	if err != nil {
 		if errors.Is(err, os.ErrNotExist) {
-			return bundle.GetKubeadminPassword()
+			return "", nil
```

Comment:
@guillaumerose this is required because https://github.com/code-ready/crc/blob/master/pkg/crc/machine/machine.go#L17 function which called https://github.com/code-ready/crc/blob/master/pkg/crc/machine/start.go#L252 and during this time cluster is not started and kubeadmin password not set so this file doesn't exist and we return empty string for password.

Reply:
`clusterConfig` in `Start()` does not seem to be needed until much later, so you could move the call to `getClusterConfig` after the generation of the kubeadmin password?

Reply:
Yes, we could construct the clusterConfig (or an other clusterConfig) at the very end of machine.Start().

Reply:
@cfergeau @guillaumerose updated, make sense thanks 👍🏼